src/sentry/monitors/serializers.py

```diff
@@ -6,22 +6,65 @@
 from sentry.api.serializers import ProjectSerializerResponse, Serializer, register, serialize
 from sentry.models import Project
 
-from .models import Monitor, MonitorCheckIn
+from .models import Monitor, MonitorCheckIn, MonitorEnvironment
+
+
+@register(MonitorEnvironment)
+class MonitorEnvironmentSerializer(Serializer):
+    def serialize(self, obj, attrs, user):
+        return {
+            "status": obj.get_status_display(),
+            "name": obj.environment.name,
```

Comment:
This will cause an n+1. I'd recommend either making a `.select_related("environment")` on the queryset you're getting these from, or add a `get_attrs` and use `prefetch_related_objects(item_list, "environment")`